src/hotspot/cpu/x86/stubGenerator_x86_64_string.cpp

```diff
@@ -0,0 +1,2585 @@
+    __ cmpq(rax, Address(rcx, 0));
+    __ jne(L_top_loop_1);
+    __ movzwl(rax, Address(rbx, r15, Address::times_1, -0x14));
+    __ cmpw(Address(r10, 0x9), rax);
+    __ jne(L_top_loop_1);
+    __ movzbl(rax, Address(rbx, r15, Address::times_1, -0x12));
+    __ cmpb(rax, Address(r10, 0xb));
+    __ jne(L_top_loop_1);
+    __ jmp(L_0x406019);
+
+    //  CASE 13:
+    small_hs_jmp_table[13] = __ pc();
+    __ movq(rax, Address(rbx, r15, Address::times_1, -0x1c));
+    __ cmpq(rax, Address(r10, 0x1));
+    __ jne(L_top_loop_1);
+    __ movl(rax, Address(r10, 0x9));
+    __ cmpl(Address(rbx, r15, Address::times_1, -0x14), rax);
+    __ jne(L_top_loop_1);
+    __ jmp(L_0x406019);
+
+    //  CASE 14:
+    small_hs_jmp_table[14] = __ pc();
+    __ movq(rax, Address(rbx, r15, Address::times_1, -0x1c));
+    __ cmpq(rax, Address(r10, 0x1));
+    __ jne(L_top_loop_1);
+    __ movl(rax, Address(r10, 0x9));
+    __ cmpl(Address(rbx, r15, Address::times_1, -0x14), rax);
+    __ jne(L_top_loop_1);
+    __ movzbl(rax, Address(r10, 0xd));
+    __ cmpb(Address(rbx, r15, Address::times_1, -0x10), rax);
+    __ jne(L_top_loop_1);
+    __ jmp(L_0x406019);
+
+    //  CASE 15:
+    small_hs_jmp_table[15] = __ pc();
+    __ movq(rax, Address(rbx, r15, Address::times_1, -0x1c));
+    __ cmpq(rax, Address(r10, 0x1));
+    __ jne(L_top_loop_1);
+    __ movl(rax, Address(r10, 0x9));
+    __ cmpl(Address(rbx, r15, Address::times_1, -0x14), rax);
+    __ jne(L_top_loop_1);
+    __ movzwl(rax, Address(r10, 0xd));
+    __ cmpw(Address(rbx, r15, Address::times_1, -0x10), rax);
+    __ jne(L_top_loop_1);
+    __ jmp(L_0x406019);
+
+    //  CASE 16:
+    small_hs_jmp_table[16] = __ pc();
+    __ movq(rax, Address(rbx, r15, Address::times_1, -0x1c));
+    __ cmpq(rax, Address(r10, 0x1));
+    __ jne(L_top_loop_1);
+    __ movl(rax, Address(r10, 0x9));
+    __ cmpl(Address(rbx, r15, Address::times_1, -0x14), rax);
+    __ jne(L_top_loop_1);
+    __ movzwl(rax, Address(r10, 0xd));
+    __ cmpw(Address(rbx, r15, Address::times_1, -0x10), rax);
+    __ jne(L_top_loop_1);
+    __ movzbl(rax, Address(r10, 0xf));
+    __ cmpb(Address(rbx, r15, Address::times_1, -0xe), rax);
+    __ jne(L_top_loop_1);
+    __ jmp(L_0x406019);
+
+    //  CASE 17:
+    small_hs_jmp_table[17] = __ pc();
+    __ movq(rax, Address(rbx, r15, Address::times_1, -0x1c));
+    __ cmpq(rax, Address(r10, 0x1));
+    __ jne(L_top_loop_1);
+    __ movq(rax, Address(r10, 0x9));
+    __ cmpq(Address(rbx, r15, Address::times_1, -0x14), rax);
+    __ jne(L_top_loop_1);
+    __ jmp(L_0x406019);
+
+    //  CASE 18:
+    small_hs_jmp_table[18] = __ pc();
+    __ movdqu(xmm0, Address(rbx, r15, Address::times_1, -0x1c));
+    __ vpsubb(xmm0, xmm0, Address(r10, 0x1), Assembler::AVX_128bit);
+    __ vptest(xmm0, xmm0, Assembler::AVX_128bit);
+    __ jne(L_top_loop_1);
+    __ movzbl(rax, Address(r10, 0x11));
+    __ cmpb(Address(rbx, r15, Address::times_1, -0xc), rax);
+    __ jne(L_top_loop_1);
+    __ jmp(L_0x406019);
```

Comment:
For 16 bytes and above can't we directly use [V]PTEST instruction to save multiple loads and compares.